scripts/mpi_cmake_modules/clang_format.py

```diff
@@ -0,0 +1,87 @@
+#!/usr/bin/env python
+"""Execute a clang-format on a desired file or folder.
+
+Example:
+    1/ rosrun mpi_cmake_module clang_format /path/to/format
+    2/ rosrun mpi_cmake_module clang_format file_to_format.ext
+"""
+
+import sys
+from os import walk
+from os import path
+import subprocess
+try:
+    from shutil import which
+except:
+    import distutils.spawn
+    which = distutils.spawn.find_executable
+import mpi_cmake_modules
+from mpi_cmake_modules.yaml2oneline import yaml2oneline
+
+
+def find_clang_format(name_list=['clang-format']):
+    """ Find the full path to the clang-format executable. """
+    for name in name_list:
+        path_to_clang_format = which(name)
+        if path_to_clang_format is not None:
+            return path_to_clang_format
+
+
+def load_clang_format_config():
+    for module_path in mpi_cmake_modules.__path__:
+        for (dirpath, _, filenames) in walk(module_path):
+            if "_clang-format" in filenames:
+                return yaml2oneline(path.join(dirpath, "_clang-format"))
+            break
+
+
+def test_valid_file(filename):
+    if path.isfile(filename):
+        if (filename.endswith(".h") or filename.endswith(".c") or
```

Comment:
As both else-cases are the same, these if's could be combined with an `and`.

Reply:
`if any([filename.endswith(suffix) for suffix in (".h",".c",".cpp")])`